blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanCredentialsProvider.java

```diff
@@ -0,0 +1,307 @@
+            this.user = user;
+            this.id = id;
+            this.domain = domain;
+            this.domainRequirements = domainRequirements;
+        }
+
+        public @Nonnull String getUser() {
+            return user;
+        }
+
+        public @Nonnull String getId() {
+            return id;
+        }
+
+        public @Nonnull String getDomain() {
+            return domain;
+        }
+
+        public @Nonnull List<DomainRequirement> getDomainRequirements() {
+            return domainRequirements;
+        }
+
+        @Override
+        public AbstractFolderProperty<?> reconfigure(StaplerRequest req, JSONObject form) throws FormException {
+            return this;
+        }
+
+        @Extension
+        public static class DescriptorImpl extends AbstractFolderPropertyDescriptor {
+        }
+
+        public static class CredentialsStoreActionImpl extends CredentialsStoreAction {
+
+            private final CredentialsStore store;
+
+            CredentialsStoreActionImpl(CredentialsStore store) {
+                this.store = store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Nonnull
+            @Override
+            public CredentialsStore getStore() {
+                return store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconFileName() {
+                return isVisible()
+                        ? "/plugin/credentials/images/48x48/folder-store.png"
+                        : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconClassName() {
+                return isVisible()
+                        ? "icon-credentials-folder-store"
+                        : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getDisplayName() {
+                return Messages.BlueOceanCredentialsProvider_DisplayName();
+            }
+
+            @Override
+            public String getUrlName() {
+                return super.getUrlName();
+            }
+        }
+
+        public static class StoreImpl extends CredentialsStore{
+            private final AbstractFolder abstractFolder;
+            private final CredentialsStoreAction storeAction;
+
+            StoreImpl(AbstractFolder abstractFolder) {
+                super(BlueOceanCredentialsProvider.class);
+                this.abstractFolder = abstractFolder;
+                this.storeAction = new CredentialsStoreActionImpl(this);
+            }
+
+            @Nonnull
+            @Override
+            public List<Domain> getDomains() {
+                FolderPropertyImpl prop = (FolderPropertyImpl) abstractFolder.getProperties().get(FolderPropertyImpl.class);
```

Comment:
No, your property should have its own domain with the domain specification that it is exposing the proxy credentials on... and that is the domain you should be returning here... not the domain object instance from the user's store